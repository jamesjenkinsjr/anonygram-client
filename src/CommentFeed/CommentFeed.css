.CommentFeed__form {
  margin: 1em;
  height: 3em;
}

.CommentFeed__input {
  width: 80%;
  height: 100%;
  padding: 1em;
}

.CommentFeed__button {
  width: 20%;
  height: 100%;
}