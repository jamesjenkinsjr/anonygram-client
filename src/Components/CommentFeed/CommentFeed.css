.CommentFeed__form {
  width: 100%;
  height: 3em;

}

.CommentFeed__input {
  width: 80%;
  height: 100%;
  padding: 1em;
  box-shadow: -2px 3.5px  #d4dbdf;
  border-radius: 5px;
}

.CommentFeed__button {
  width: 20%;
  height: 100%;
  background-color: #03a9f4;
  box-shadow: -2px 3.5px  #007ac1;
  padding: 1em;
  border-radius: 5px;
}


