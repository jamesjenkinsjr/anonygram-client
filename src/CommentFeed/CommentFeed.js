import React, { Component } from 'react';
import randomizer from '../helpers/randomizer';
import Comment from './Comment/Comment';
import './CommentFeed.css';

export default class CommentFeed extends Component {

  state = {
    usernames: {},
    incrementor: -1,
  }

  componentDidMount() {
    const newUsernames = randomizer.getRandomUsernames(this.props.comments);
    this.setState({ usernames: newUsernames });
  };

  render() {
    const { comments } = this.props;
    console.log(this.state.usernames)
    console.log(this.state.usernames[1])
    return (
      <div>
        {comments.map(commentObj => {
          const { userId, id, text } = commentObj;
          return <Comment key={id} text={text} username={this.state.usernames[userId]} />;
        })}
        <form>
          <input />
          <button>Post</button>
        </form>
      </div>
    )
  }
}

