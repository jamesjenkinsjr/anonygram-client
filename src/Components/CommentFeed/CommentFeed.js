import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import randomizer from '../../helpers/randomizer';
import Comment from './Comment/Comment';
import CommentApi from '../../services/comment-api-service';
import UserContext from '../../contexts/UserContext';
import './CommentFeed.css';

import moment from 'moment';

export default class CommentFeed extends Component {
  state = {
    usernames: [],
    newComment: '',
  }
  
  static contextType = UserContext;

  processTimestamp(timestamp) {
    return moment(new Date(timestamp)).local().fromNow();
  }

  handleSubmit = (e) => {
    e.preventDefault();
    const commentText = e.target.newComment.value; 
    CommentApi.postComment(this.props.id, commentText, this.context.user.id)
      .then(res => {
        console.log(res)
        this.props.setCommentsByPush(res);
        this.setState({ newComment: '' })
      })
  }

  componentDidMount() {
    console.log(this.context.user.id)
    if (this.props.comments.length > 1) {
      this.generateUsernames(this.props.comments)
    }
  };

  generateUsernames = (comments) => {
    const newUsernames = randomizer.getAnonUsernames(comments);
    this.setState({ usernames: newUsernames });
  }

  render() {
    const { comments } = this.props;

    //If you're posting the first comment, then generateUsernames didn't run on mount. Run it here. ComponentDidMount edited to be >1 so that generateUsernames doesn't run twice if there is one comment to begin with
    (this.props.comments.length === 1) && this.generateUsernames(this.props.comments);
    
    return (
      <div>
        {(comments.length === 0) 
          ? null
          : (
            comments.map(commentObj => {
              const { comment_timestamp, user_id, comment_id, comment_text } = commentObj;
    
              return <Comment key={comment_id} text={comment_text} username={this.state.usernames[user_id]} timestamp={this.processTimestamp(comment_timestamp)} />;
            })
          )
        }
        <form onSubmit={(e) => this.handleSubmit(e)} className='CommentFeed__form'>
          {/* <label htmlFor='newComment'>Add a Comment</label> */}
          { //Conditionally render for logged in user
            (this.context.user.id !== '') ? (
              <>
                <input type='text' id='newComment' className='CommentFeed__input' placeholder='Add a comment...' />
                <button type='submit' className='CommentFeed__button'>Post</button>
              </>
            ) : (
              <>
                <label htmlFor='newComment'>You must be logged in to post a comment. Log in <Link to='/login'>here</Link>.</label>
                <input type='text' id='newComment' className='CommentFeed__input' placeholder='' />
                <button type='submit' className='CommentFeed__button' disabled>Post</button>
              </>
            )
          }
        </form>
      </div>
    )
  }
}

