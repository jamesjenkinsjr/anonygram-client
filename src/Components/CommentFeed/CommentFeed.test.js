import React from 'react';
import ReactDOM from 'react-dom';
import CommentFeed from './CommentFeed';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CommentFeed />, div);
  ReactDOM.unmountComponentAtNode(div);
});
