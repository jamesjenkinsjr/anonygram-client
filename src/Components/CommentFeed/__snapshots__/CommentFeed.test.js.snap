// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommendFeed component renders elements by default 1`] = `
<div>
  <form
    className="CommentFeed__form"
    onSubmit={[Function]}
  >
    <input
      className="CommentFeed__input"
      id="newComment"
      onChange={[Function]}
      placeholder="Add a comment..."
      type="text"
      value=""
    />
    <button
      className="CommentFeed__button"
      type="submit"
    >
      Post
    </button>
  </form>
</div>
`;

exports[`CommendFeed component renders elements given props 1`] = `
<div>
  <Comment
    key="56532f9e-8022-4759-b134-b7269d510689"
    text="comment text 1"
    timestamp="a few seconds ago"
  />
  <form
    className="CommentFeed__form"
    onSubmit={[Function]}
  >
    <input
      className="CommentFeed__input"
      id="newComment"
      onChange={[Function]}
      placeholder="Add a comment..."
      type="text"
      value=""
    />
    <button
      className="CommentFeed__button"
      type="submit"
    >
      Post
    </button>
  </form>
</div>
`;
